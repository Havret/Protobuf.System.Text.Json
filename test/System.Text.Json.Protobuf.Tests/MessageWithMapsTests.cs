using System.Text.Json.Protobuf.Tests.Utils;
using Shouldly;
using SmartAnalyzers.ApprovalTestsExtensions;
using Xunit;

namespace System.Text.Json.Protobuf.Tests;

public class MessageWithMapsTests
{
    [Fact]
    public void Should_serialize_message_with_map_field()
    {
        // Arrange
        var msg = new MessageWithMaps
        {
            MapIntToString = {[1] = "1-value"},
            MapStringToComplexType =
            {
                ["string_key"] = new NestedMessageAsKey
                {
                    MapStringToInt = {["a"] = 1}
                }
            }
        };
        var jsonSerializerOptions = TestHelper.CreateJsonSerializerOptions();

        // Act
        var serialized = JsonSerializer.Serialize(msg, jsonSerializerOptions);

        // Assert
        var approver = new ExplicitApprover();
        approver.VerifyJson(serialized);
    }

    [Fact]
    public void Should_deserialize_message_with_map_field()
    {
        // Arrange
        var msg = new MessageWithMaps
        {
            MapIntToString = {[1] = "1-value"},
            MapStringToComplexType =
            {
                ["string_key"] = new NestedMessageAsKey
                {
                    MapStringToInt = {["a"] = 1}
                }
            }
        };
        var jsonSerializerOptions = TestHelper.CreateJsonSerializerOptions();

        // Act
        var serialized = JsonSerializer.Serialize(msg, jsonSerializerOptions);
        var deserialized = JsonSerializer.Deserialize<MessageWithMaps>(serialized, jsonSerializerOptions);

        // Assert
        deserialized.ShouldNotBeNull();
        deserialized.ShouldBeEquivalentTo(msg);
    }
}