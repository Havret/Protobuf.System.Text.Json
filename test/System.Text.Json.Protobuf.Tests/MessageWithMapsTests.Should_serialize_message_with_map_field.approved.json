{
  "mapIntToString": {
    "1": "1-value"
  },
  "mapStringToComplexType": {
    "string_key": {
      "mapStringToInt": {
        "a": 1
      }
    }
  }
}